yarn/src/main/scala/org/apache/spark/deploy/yarn/ClientArguments.scala

```diff
@@ -87,6 +92,21 @@ private[spark] class ClientArguments(args: Array[String], sparkConf: SparkConf)
       throw new IllegalArgumentException(
         "You must specify at least 1 executor!\n" + getUsageMessage())
     }
+    if (isClusterMode) {
+      for (key <- Seq(amMemKey, amMemOverheadKey)) {
+        if (sparkConf.getOption(key).isDefined) {
+          println(s"$key is set but does not apply in cluster mode.")
+        }
+      }
+      amMemory = driverMemory
+    } else {
+      for (key <- Seq(driverMemKey, driverMemOverheadKey)) {
+        if (sparkConf.getOption(key).isDefined) {
+          println(s"$key is set but does not apply in client mode.")
+        }
+      }
+      amMemory = Utils.memoryStringToMb(sparkConf.get(amMemKey, "512m"))
```

Comment:
The default value of 512 is duplicated here and up there. I would do this instead
```
sparkConf.getOption(amMemKey)
  .map(Utils.memoryStringToMb)
  .foreach { mem => amMemory = mem }
```